test/extended/olm/olmv1.go

```diff
@@ -419,3 +500,89 @@ func checkFeatureCapability(ctx context.Context, oc *exutil.CLI) {
 		g.Skip("Test only runs with OperatorLifecycleManagerV1 capability")
 	}
 }
+
+// verifyAPIEndpoint runs a job to validate the given service endpoint of a ClusterCatalog
+func verifyAPIEndpoint(ctx g.SpecContext, oc *exutil.CLI, namespace, catalogName, endpoint string) {
+	jobName := fmt.Sprintf("test-catalog-%s-%s-%s", catalogName, endpoint, rand.String(5))
+
+	baseURL, err := oc.AsAdmin().Run("get").Args(
+		"clustercatalogs.olm.operatorframework.io",
+		catalogName,
+		"-o=jsonpath={.status.urls.base}").Output()
+	o.Expect(err).NotTo(o.HaveOccurred())
+	o.Expect(baseURL).NotTo(o.BeEmpty(), fmt.Sprintf("Base URL not found for catalog %s", catalogName))
+
+	serviceURL := fmt.Sprintf("%s/api/v1/%s", baseURL, endpoint)
+	g.GinkgoLogr.Info(fmt.Sprintf("Using service URL: %s", serviceURL))
+
+	job := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      jobName,
+			Namespace: namespace,
+		},
+		Spec: batchv1.JobSpec{
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  "api-tester",
+							Image: "registry.redhat.io/rhel8/httpd-24:latest",
+							Command: []string{
+								"/bin/bash",
+								"-c",
+								fmt.Sprintf(`
+set -ex
+response=$(curl -s -k "%s" || echo "ERROR: Failed to access endpoint")
+if [[ "$response" == ERROR* ]]; then
+  echo "$response"
+  exit 1
+fi
+echo "$response" > /tmp/api-response
+
+# check if response can be parsed as new line delimited JSON
+if cat /tmp/api-response | jq -s . > /dev/null 2>&1; then
+  echo "Valid JSON response detected"
+  exit 0
+fi
+echo "ERROR: Invalid JSON response"
+exit 1
+`, serviceURL),
+							},
+						},
+					},
+					RestartPolicy: corev1.RestartPolicyNever,
+				},
+			},
+		},
+	}
+
+	_, err = oc.AdminKubeClient().BatchV1().Jobs(namespace).Create(context.TODO(), job, metav1.CreateOptions{})
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	err = wait.PollUntilContextTimeout(ctx, 5*time.Second, 30*time.Second, true, func(ctx context.Context) (bool, error) {
+		job, err := oc.AdminKubeClient().BatchV1().Jobs(namespace).Get(context.TODO(), jobName, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+
+		if job.Status.Succeeded > 0 {
+			return true, nil
+		}
+		if job.Status.Failed > 0 {
+			return false, fmt.Errorf("job failed")
+		}
```

Comment:
Looking at the Job conditions, it seems like we can check for `type: Complete` (success) or `type: Failed`(failed). Maybe we should also consider looking at the difference between `status.completionTime` and `status.startTime` for reporting/checking duration.